broker/src/test/java/io/camunda/zeebe/broker/transport/backupapi/BackupApiRequestHandlerTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.transport.backupapi;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.camunda.zeebe.logstreams.log.LogStreamRecordWriter;
+import io.camunda.zeebe.protocol.impl.encoding.AdminResponse;
+import io.camunda.zeebe.protocol.impl.encoding.BackupRequest;
+import io.camunda.zeebe.protocol.impl.encoding.ErrorResponse;
+import io.camunda.zeebe.protocol.record.BackupRequestType;
+import io.camunda.zeebe.protocol.record.ErrorCode;
+import io.camunda.zeebe.scheduler.testing.ControlledActorSchedulerExtension;
+import io.camunda.zeebe.transport.ServerOutput;
+import io.camunda.zeebe.transport.impl.AtomixServerTransport;
+import io.camunda.zeebe.util.Either;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import org.agrona.ExpandableArrayBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class BackupApiRequestHandlerTest {
+
+  @RegisterExtension
+  ControlledActorSchedulerExtension scheduler = new ControlledActorSchedulerExtension();
+
+  final AtomixServerTransport transport = mock(AtomixServerTransport.class);
+  BackupApiRequestHandler handler;
+  private final LogStreamRecordWriter logStreamRecordWriter = mock(LogStreamRecordWriter.class);
+  private ServerOutput serverOutput;
+
+  private CompletableFuture<Either<ErrorResponse, AdminResponse>> responseFuture;
+
+  @BeforeEach
+  void setup() {
+    when(logStreamRecordWriter.metadataWriter(any())).thenReturn(logStreamRecordWriter);
```

Comment:
:wrench: I think you wanted to update this: https://github.com/camunda/zeebe/pull/9899#discussion_r932099368